example/src/main/scala/ComplexExampleMain.scala

```diff
@@ -0,0 +1,63 @@
+import cats.effect.{IO, Sync}
+import cats.implicits._
+import org.typepure.uuid4s.{FUUID, UUID}
+import cats.Id
+
+case class ProfileResponse(userId: String,
+                           email: String,
+                           companyId: String,
+                           companyName: String,
+                           photo: String)
+
+case class Company(id: Id[UUID], name: String)
+
+case class User(id: Id[UUID], email: String)
+
+case class Profile(user: User, company: Company, photo: String)
+
+class ProfileConverter {
+  def fromF[F[_]: Sync: FUUID](profileResponse: ProfileResponse): F[Profile] = {
```

Comment:
An alternative apporach would be:

```
def fromF[F[_]: Sync: FUUID](profileResponse: ProfileResponse)(implcit fuuid: FUUID[F]): F[Profile] = 
{
    for {
      userId <- fuuid.fromString(profileResponse.userId)
      companyId <- fuuid.fromString(profileResponse.companyId)
    } yield ....
}

```

Reply:
Yes, check the countext bound in scala: https://docs.scala-lang.org/tutorials/FAQ/context-bounds.html